examples/consul-ami/consul.json

```diff
@@ -51,7 +51,7 @@
   },{
     "type": "shell",
     "inline": [
-      "if [[ -n '{{user `download_url`}}' ]]; then",
+      "if test -n \"{{user `download_url`}}\"; then",
```

Comment:
Why the change?

Reply:
That... sounds very odd.

BTW, as I mentioned in my other PR, it might be worth allowing both `--download-url` and `--version` to be specified, but if the option is set to an empty string, to just ignore it. That would remove the need for an if-statement here.

Reply:
Oh that was the original plan in fact, but after discussing with Josh we came to the conclusion that this makes it clearear to users that these two settings are independent and one of them is required

Reply:
Right, I still think that would be the case. If you specify:

```
--download-url "foo" --version "bar"
```

You'd get an error. But if you specify:

```
--download-url "foo" --version ""
```

That is, an empty string for one of the params, it's ignored, and the other is used. 

And if you specify:

```
--download-url "" --version ""
```

The defaults are used.

Reply:
But the defaults are specified in the packer template, not in the install script.

Reply:
You can already pass both of them to the script, it is not clear from the template which one takes precedence. This is why this if is there. To make it clear.